# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.


### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [6]:
import pandas as pd
import numpy as np
from scipy import stats

In [7]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')
#data.to_csv('data/us_job_market_discrimination.csv')


In [8]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [9]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [10]:
data.describe()

education       ofjobs     yearsexp       honors    volunteer  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      3.618480     3.661396     7.842916     0.052772     0.411499   
std       0.714997     1.219126     5.044612     0.223601     0.492156   
min       0.000000     1.000000     1.000000     0.000000     0.000000   
25%       3.000000     3.000000     5.000000     0.000000     0.000000   
50%       4.000000     4.000000     6.000000     0.000000     0.000000   
75%       4.000000     4.000000     9.000000     0.000000     1.000000   
max       4.000000     7.000000    44.000000     1.000000     1.000000   

          military     empholes  occupspecific   occupbroad  workinschool  \
count  4870.000000  4870.000000    4870.000000  4870.000000   4870.000000   
mean      0.097125     0.448049     215.637782     3.481520      0.559548   
std       0.296159     0.497345     148.127551     2.038036      0.496492   
min       0.000000     0.000000       7.000000     1.000000      0.000000   
25%       0.000000     0.000000      27.000000     1.000000      0.000000   
50%       0.000000     0.000000     267.000000     4.000000      1.000000   
75%       0.000000     1.000000     313.000000     6.000000      1.000000   
max       1.000000     1.000000     903.000000     6.000000      1.000000   

          ...           educreq      compreq       orgreq        manuf  \
count     ...       4870.000000  4870.000000  4870.000000  4870.000000   
mean      ...          0.106776     0.437166     0.072690     0.082957   
std       ...          0.308866     0.496083     0.259649     0.275854   
min       ...          0.000000     0.000000     0.000000     0.000000   
25%       ...          0.000000     0.000000     0.000000     0.000000   
50%       ...          0.000000     0.000000     0.000000     0.000000   
75%       ...          0.000000     1.000000     0.000000     0.000000   
max       ...          1.000000     1.000000     1.000000     1.000000   

          transcom     bankreal        trade   busservice   othservice  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      0.030390     0.085010     0.213963     0.267762     0.154825   
std       0.171677     0.278932     0.410141     0.442847     0.361773   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     1.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

           missind  
count  4870.000000  
mean      0.165092  
std       0.371308  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000  

[8 rows x 55 columns]

In [11]:
list(data)


['id',
 'ad',
 'education',
 'ofjobs',
 'yearsexp',
 'honors',
 'volunteer',
 'military',
 'empholes',
 'occupspecific',
 'occupbroad',
 'workinschool',
 'email',
 'computerskills',
 'specialskills',
 'firstname',
 'sex',
 'race',
 'h',
 'l',
 'call',
 'city',
 'kind',
 'adid',
 'fracblack',
 'fracwhite',
 'lmedhhinc',
 'fracdropout',
 'fraccolp',
 'linc',
 'col',
 'expminreq',
 'schoolreq',
 'eoe',
 'parent_sales',
 'parent_emp',
 'branch_sales',
 'branch_emp',
 'fed',
 'fracblack_empzip',
 'fracwhite_empzip',
 'lmedhhinc_empzip',
 'fracdropout_empzip',
 'fraccolp_empzip',
 'linc_empzip',
 'manager',
 'supervisor',
 'secretary',
 'offsupport',
 'salesrep',
 'retailsales',
 'req',
 'expreq',
 'comreq',
 'educreq',
 'compreq',
 'orgreq',
 'manuf',
 'transcom',
 'bankreal',
 'trade',
 'busservice',
 'othservice',
 'missind',
 'ownership']

In [12]:
#First we need to find the proportions of b's and w's who received a call

In [13]:
#Number of black-sounding names that received a call
sum(data[data.race=='b'].call)

157.0

In [14]:
#Number of white-sounding names that received a call
sum(data[data.race=='w'].call)

235.0

In [15]:
#Number of total names in each category
data.race.value_counts()

b    2435
w    2435
Name: race, dtype: int64

In [16]:
#Total number of samples
data.id.count()

4870

# 1. What test is appropriate for this problem? Does CLT apply?

To determine whether there is a statistical significance between the proportion of blacks who received a call and the proportion of whites who received a call, we will use the central limit theorem and hypothesis testing to determine if these proportions could have been arrived at by chance.  

The central limit theorem applies because we are using a sample of resumes sent to a sample of employers to see if a resume will generate a call.  Our calculations will determine if this result could have been arrived at by chance, or if there is definitely a statistically significant difference between the two proportions.



## 2. What are the null and alternate hypotheses?

For this problem, our null hypothesis will be, "There is no difference between the two proportions".

The alternative hypothesis will be "There is a difference."

## 3. Compute margin of error, confidence interval, and p-value.

From our data:

$n_b = 2435$

$n_w = 2435$

$\bar{p}_b = \frac{157}{2435} = 0.064  $

$\bar{p}_w = \frac{235}{2435} = 0.097 $

$\bar{p}_w - \bar{p}_b = 0.097 - 0.064 = 0.033 $

To find a 95% confidence interval around the difference of proportions (0.033), we will construct a sampling distribution of the difference of proportions $\bar{p}_w - \bar{p}_b$

Our sampling standard distribution is thus:
$$ \sigma_{\bar{p}_w - \bar{p}} = \sqrt{\frac{p_w(1-p_w)}{n_w} + \frac{p_b(1-p_b)}{n_b}} $$

Since we do not have the population $p_w$ and $p_b$, we will approximate them using the sample proportions from our data.

$$ \sigma_{\bar{p}_w - \bar{p}} \approx \sqrt{\frac{0.097(1-0.097)}{2435} + \frac{0.064(1-0.064)}{2435}} = 0.0078 $$

Using the empirical rule, 95% means 2 standard deviations away from the mean, so:

$ 2 * 0.0078 = 0.0156 $

Our margin of error for the difference in proportions is $0.033 \pm 0.0156$ and our confidence interval is 0.0174 to 0.0486 



To find our p-value, we will run our hypothesis test.  

If we assume that there is no difference between the two proportions, or $p_w - p_b = 0$ we will compute the probability that we can get the value of 0.033 that we got from our data.

Our z-score then, will be:

$$z = \frac{observed mean - assumed mean}{\sigma_{\bar{p}_w - \bar{p}_b}} $$

Because we are assuming that the proportions are equal, we cannot use the approximated value of $\sigma_{\bar{p}_w - \bar{p}_b}$ that we computed above.  Instead, we will assume that $p_w = p_b$.  (Also, in this example, $n_w = n_b$)

$$\sigma_{\bar{p}_w - \bar{p}_b} = \sqrt{\frac{p_w(1-p_w)}{n_w} + \frac{p_b(1-p_b)}{n_b}} = \sqrt{\frac{p(1-p)}{n} + \frac{p(1-p)}{n}} = \sqrt{\frac{2p(1-p)}{n}} $$

For p, we will use a combined $\bar{p}$ which we will derive by combining both proportions into one sample:
$$ \bar{p} = \frac{157 + 235}{2435 + 2435} = 0.080 $$

Which we will plug into our formula for $\sigma_{\bar{p}_w - \bar{p}_b}$: 
$$\sqrt{\frac{2*0.080(1-0.080)}{2435}} = 0.0078 $$

Which we will then plug into our formula the z-score:
$$z = \frac{0.033 - 0}{0.0078} = 4.23 $$

From this z-score, we get a p-value of 1 - 0.999988 = 0.000012.

# 4. Write a story describing the statistical significance in the context or the original problem.

Our computed confidence interval, 0.0174 to 0.0486, indicates that there IS at least a 0.0174 difference in the proportions of calls received by resumes with white-sounding versus black-sounding names.  Our hypothesis test, and our really small p-value indicates that there is a difference between the proportion of calls received by white-sounding names and black-sounding names.


# 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

This analysis does NOT indicate that race/name is the MOST important factor.  However it does indicate that it IS a factor, and resumes with black-sounding names receive fewer calls than resumes with white-sounding names.

Because the names on the resumes were randomized, we have ruled out other factors such as their background, previous work experience, and schooling.  

Further analysis should further break down the sample data into other categories, such as gender, and the city where the employer is located (Boston or Chicago).
